crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -56,6 +56,9 @@ pub struct Interaction {
 pub struct Pool {
     pub reserve_a: Asset,
     pub reserve_b: Asset,
+    /// Specifies a predefined quote. Original AMM formula is used in case the
+    /// quote is empty.
+    pub liquidity_quote: Option<LiquidityQuote>,
```

Comment:
The initial idea was to get rid of the AMM formula and to explicitly provide a quote here:
https://github.com/cowprotocol/services/blob/9db38e05afa265cb40dcffef265f9679becb4c48/crates/driver/src/tests/setup/blockchain.rs#L79-L85
That frees us from caring about the AMM formula at all. The pool is adjusted "under the hood" just to make the simulation pass with our custom quotes since we continue using the uniswap contract. A single pool can be used for multiple orders. Don't really see other options here.

Reply:
Another option is to introduce an optional `buy_amount` field in `Order` struct. If the value is specified, then `Pool` uses this value instead of calculating with AMM formula. But in this case, the pool can be used only for a single order. If any other order is added with a different sell/buy ratio, on-chain simulation will fail.

Reply:
Why can we not use `order.sell_amount` like we do when no quote is specified? Assuming that the reserves have been adjusted for the quote, when trying to trade `sell_amount` with the pool we should get exactly quote.buy_amount & quote.sell_amount back.

What am I missing?

Reply:
Like that? https://github.com/cowprotocol/services/pull/2404/commits/c4810f6aa1c255892b1876d8805ca5ed2046c52e
Initially, I just wanted to avoid using the AMM formula when the adjusted pool reserves. Do we now need to specify the order's `buy_amount` explicitly? If yes, how should it be used? The AMM's `out` function result should be compared with the `order.buy_amount`?